controller.go

```diff
@@ -157,11 +160,23 @@ func New(cfgOptions ...config.Option) (NetworkController, error) {
 			// But it cannot fail creating the Controller
 			log.Debugf("Failed to Initialize Discovery : %v", err)
 		}
+		if err := c.initLocalStore(); err != nil {
+			return nil, fmt.Errorf("Failed to Initialize LocalDatastore due to %v.", err)
+		}
 	}
 
 	return c, nil
 }
 
+// Start method starts the controller
+func (c *controller) Start() error {
+	if c.started {
```

Comment:
Can there be any concurrent call to the method `Start()`, also exercisable via an ad-hoc test ?
Just wondering whether locking would be justifiable here.